v9.1 changelog & docs (#6662)
mattermost/docs: source/install/self-managed-changelog.md

### Compatibility
 - Updated Chromium minimum supported version to 116+.

### Improvements

#### User Interface (UI)
 - Added a **Cancel** button to the **Delete category** modal.
 - Added the ability to resize the channel sidebar and right-hand sidebar.
 - Added two new filtering options (show all channel types and show private channels) to the **Browse channels** modal.
 - Pre-packaged GitLab plugin version [v1.7.0](https://github.com/mattermost/mattermost-plugin-gitlab/releases/tag/v1.7.0).
 - Pre-packaged Calls version v0.19.1.
 - Added additional reaction options when viewing threads or messages when the sidebar is larger than its minimum width.
 - Added a link to [notification documentation](https://docs.mattermost.com/preferences/manage-your-notifications.html) in the **Notification Settings** modal.
 - Updated the post textbox measurement code to be more reliable.
 - The ``/invite`` slash command now supports custom user groups.
 - Re-enabled the remote marketplace functionality, when configured as per ``PluginSettings.EnableRemoteMarketplace`` [documentation](https://docs.mattermost.com/configure/plugins-configuration-settings.html#plugins-enableremotemarketplace).

>  - Added the ability to convert a group message channel to a private channel.

Reviewer: "Added the ability to convert a group message to a private channel."

>  - Group message channels now behave like direct message channels, showing a new mention for every new message.

Reviewer: **Change to**:
Group messages (GMs) now behave like direct messages (DMs). The badge count increases for every new message.

**Context**:
- Recommend we use this opportunity to clearly differentiate channels vs DMs/GMs from how we describe it to users. 
- Also, let's move away from "mention badge", instead opting for "badge". For example, DMs trigger a notification (aka "trigger a mention") but the user didn't explicitly use the @ mention.
- There's [efforts](https://docs.google.com/document/d/1h3Dygq0cpZ8eGfCItIqf_5clhXuhkoHRP4mSeIY1-RI/edit#heading=h.fqqw3aagrxyi) underway re: updating naming

cc: @cwarnermm in case you have more input

**Release day: October 16, 2023**

### Important Upgrade Notes
 - Improved performance on data retention ``DeleteOrphanedRows`` queries. See the [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html) for notes on a new migration that was added. Removed feature flag ``DataRetentionConcurrencyEnabled``. Data retention now runs without concurrency in order to avoid any performance degradation. Added a new configuration setting ``DataRetentionSettings.RetentionIdsBatchSize``, which allows admins to to configure how many batches of IDs will be fetched at a time when deleting orphaned reactions. The default value is 100.

**IMPORTANT:** If you upgrade from a release earlier than v9.0, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).

### Compatibility
 - Updated Chromium minimum supported version to 116+.

### Improvements

#### User Interface (UI)
 - Added a **Cancel** button to the **Delete category** modal.
 - Added the ability to resize the channel sidebar and right-hand sidebar.

>  - Added two new filtering options (show all channel types, show private channels) to the **Browse channels** modal.

Reviewer 2: ```suggestion
 - Added two new filtering options (show all channel types and show private channels) to the **Browse channels** modal.
```

 - Pre-packaged GitLab plugin version v1.7.0.
 - Added additional reaction options when viewing threads or messages when the sidebar is larger than its minimum width.

>  - Added block changes to name, display name, and purpose for direct and group messages.

Reviewer 3: I was trying to look for changes in the interface, and I was unable to understand what this means and how do I access that. Probably a better description of this change is needed here to make it understandable for every body.

Author: The release note comes from https://github.com/mattermost/mattermost/pull/24199. @larkox @johndavidlugtu is there a better way to describe this change?

Reviewer 4: Maybe something in the lines of:

- Improve the API to block any changes to direct and group messages name, display name or purpose. (since those are not meant to be changed)

Reviewer 3: Ok, then I think that shouldn't be in the "UI" section, because that is not a UI related thing, it is affecting only the backend (AFAIK).

Author: I've moved it to the Bug Fixes section as I believe it was a bug fix.

 - Added a link to [notification documentation](https://docs.mattermost.com/preferences/manage-your-notifications.html) in the **Notification Settings** modal.
 - Updated the post textbox measurement code to be more reliable.

>  - Pre-packaged Calls version v0.19.1.

Reviewer 2: Suggestion: group these pre-packaged bullet points together, one by one. Might make them a bit easier to skim/absorb.

>  - The ``invite`` slash command now supports custom user groups.

Reviewer 2: ```suggestion
 - The ``/invite`` slash command now supports custom user groups.
```

 - Re-enabled the remote marketplace functionality, when configured as per ``PluginSettings.EnableRemoteMarketplace`` [documentation](https://docs.mattermost.com/configure/plugins-configuration-settings.html#plugins-enableremotemarketplace).
 - Added the ability to convert a group message channel to a private channel.
 - Group message channels now behave like direct message channels, showing a new mention for every new message.

#### Administration

>  - Added ``mattermost-plugin-api`` into the mono repo.

Reviewer 2: Consider swapping out "mono repo" for the official name of the mattermost repo for discoverability. ("mono repo" is generally an internal term that may not be well understood externally).

 - Updated the public server module version to v0.0.9.
 - Added 2 new URL parameters to ``GET /api/v4/groups``: ``include_archived`` and ``filter_archived``. Added the ability to restore archived groups from the user groups modal.
 - Added file storage information to the support package.
 - A ``user_id`` is now included in all HTTP logs (debug level) to help determine who is generating unexpected traffic.
 - Added new URL Parameter to ``GET /api/v4/groups`` and ``GET /api/v4/groups/:group_id``. ``include_member_ids`` will add all the members ``user_ids`` to the group response objects. You can now also add group members to a channel, any members that are not part of the team can be added to the team through this flow and subsequently added the channel.

#### Plugin Changes
 - Added new frontend plugin extension point for the new messages separator bar.
 - Added a new plugin extensibility point to add actions to the code blocks.
 - Added the plugin hook ``UserHasBeenDeactivated``.

>  - Added a new server side plugin API method to set the searchable content for a file info (``SetFileSearchableContent``). The ``MessageHasBeenPosted`` plugin hook is now executed after the attachments are linked to the post.

Reviewer 2: "a file info" reads a bit strangely. Is the "a" needed for comprehension?

### Bug Fixes
 - Fixed keyboard support for the left-hand side channel menu, the left-hand side category menu, and the post dot menu.
 - Fixed display name in the ``comment_on`` component.

>  - Fixed an issue with keyboard support for some menus with submenus.

Reviewer 2: Does this bullet point overlap with the first bullet point in this Bug Fixes section?

>  - Fixed an issue with a disappearing punctuation when following a group mention.

Reviewer 2: ```suggestion
 - Fixed an issue with disappearing punctuation when following a group mention.
```

Reviewer 2: This bug intrigues me :) Disappearing punctuation you say!

 - Under ``DisplaySettings`` in ``config.json``:
    - Added a setting ``MaxMarkdownNodes`` to limit the maximum complexity of markdown text on mobile.

 #### Changes to Enterprise plan:
 - Under ``DataRetentionSettings`` in ``config.json``:
    - Added a new configuration setting ``RetentionIdsBatchSize``, which allows admins to to configure how many batches of IDs will be fetched at a time when deleting orphaned reactions. The default value is 100.

### API Changes
 - Added the ``X-Forwarded-For`` request header to the audit stream for all Rest API calls.
 - Added API endpoint ``POST /api/v4/user/login/desktop_login``. Modified OAuth/SAML flows to include ``desktop_login`` where applicable.
 - Added new API endpoint ``GET`` ``/api/v4/channels/<channel-id>/common_teams`` to fetch list of teams common between members of a group message.
 - Added new API endpoint ``POST`` ``/api/v4/channels/<channel-id>/convert_to_channel`` to convert a group message to a private channel.
 - Added a new ``MessageHasBeenDeleted`` hook to the plugin API.
 - Moved the ``request`` package into the public shared folder.

### Go Version

>  - v9.1 is built with Go ``v1.20.0``.

Reviewer 5: 1.20.7. Not sure from where you got the incorrect patch release version.

Reviewer 5: Ah okay, I thought I had mentioned this, but the file to look for the exact version is `server/build/Dockerfile.buildenv`.

### Known Issues

>  - Known issues related to the new feature to convert a Group Message channel to a private channel: [MM-54525](https://mattermost.atlassian.net/browse/MM-54525), [MM-54526](https://mattermost.atlassian.net/browse/MM-54526), [MM-54541](https://mattermost.atlassian.net/browse/MM-54541), [MM-54542](https://mattermost.atlassian.net/browse/MM-54542).

Reviewer 6: https://mattermost.atlassian.net/browse/MM-54525 - Is a mobile issue, do we want to add in the mobile release notes instead?

https://mattermost.atlassian.net/browse/MM-54541 and https://mattermost.atlassian.net/browse/MM-54542 - has a fix that seems that will be cherrypicked so we can probably remove?

 - ``/invite`` user to a channel incorrectly posts an ephemeral message of ``<no value> added to {channel_name} channel`` [MM-54555](https://mattermost.atlassian.net/browse/MM-54555).
 - Left-hand side resize option overrides the **Browse/Create Channel** menu if To-Do plugin is installed [MM-54367](https://mattermost.atlassian.net/browse/MM-54367).
 - Copy pasting images from Chrome fails [MM-54486](https://mattermost.atlassian.net/browse/MM-54486).
 - Adding an @mention at the start of a post draft and pressing the left or right arrow key can clear the post draft and the undo history [MM-33823](https://mattermost.atlassian.net/browse/MM-33823).
 - Google login fails on the Classic mobile apps.
 - Status may sometimes get stuck as **Away** or **Offline** in High Availability mode with IP Hash turned off.
 - Searching stop words in quotation marks with Elasticsearch enabled returns more than just the searched terms.

>  - The team sidebar on the desktop app does not update when channels have been read on mobile.

Reviewer 6: @amyblais Just for reference, can you point me to the ticket for this issue?

Author: https://mattermost.atlassian.net/browse/MM-7404, but it's closed.

 - Pre-packaged Calls plugin version [v0.20.0](https://github.com/mattermost/mattermost-plugin-calls/releases/tag/v0.20.0).
 - Pre-packaged Playbooks version [v1.39.0](https://github.com/mattermost/mattermost-plugin-playbooks/releases/tag/v1.39.0).
 - Added additional reaction options when viewing threads or messages when the sidebar is larger than its minimum width.
 - Added a link to [notification documentation](https://docs.mattermost.com/preferences/manage-your-notifications.html) in the **Notification Settings** modal.
 - Updated the post textbox measurement code to be more reliable.
 - The ``/invite`` slash command now supports custom user groups.
 - Re-enabled the remote marketplace functionality, when configured as per ``PluginSettings.EnableRemoteMarketplace`` [documentation](https://docs.mattermost.com/configure/plugins-configuration-settings.html#plugins-enableremotemarketplace).
 - Added the ability to convert a group message to a private channel.
 - Group messages (GMs) now behave like direct messages (DMs). The badge count increases for every new message.

#### Administration

>  - Added ``mattermost-plugin-api`` into the mattermost repo.

Reviewer 2: ```suggestion
 - Added ``mattermost-plugin-api`` into the ``mattermost`` GitHub repository.
```

 - Updated the public server module version to v0.0.9.
 - Added 2 new URL parameters to ``GET /api/v4/groups``: ``include_archived`` and ``filter_archived``. Added the ability to restore archived groups from the user groups modal.
 - Added file storage information to the support package.
 - A ``user_id`` is now included in all HTTP logs (debug level) to help determine who is generating unexpected traffic.

>  - Added new URL Parameter to ``GET /api/v4/groups`` and ``GET /api/v4/groups/:group_id``. ``include_member_ids`` will add all the members ``user_ids`` to the group response objects. You can now also add group members to a channel, any members that are not part of the team can be added to the team through this flow and subsequently added the channel.

Reviewer 2: ```suggestion
 - Added new URL parameter to ``GET /api/v4/groups`` and ``GET /api/v4/groups/:group_id``. ``include_member_ids`` will add all the members ``user_ids`` to the group response objects. You can now also add group members to a channel, any members that are not part of the team can be added to the team through this flow and subsequently added the channel.
```

 - Under ``DisplaySettings`` in ``config.json``:
    - Added a setting ``MaxMarkdownNodes`` to limit the maximum complexity of markdown text on mobile.

 #### Changes to Enterprise plan:
 - Under ``DataRetentionSettings`` in ``config.json``:
    - Added a new configuration setting ``RetentionIdsBatchSize``, which allows admins to to configure how many batches of IDs will be fetched at a time when deleting orphaned reactions. The default value is 100.

### API Changes
 - Added the ``X-Forwarded-For`` request header to the audit stream for all Rest API calls.
 - Added API endpoint ``POST /api/v4/user/login/desktop_login``. Modified OAuth/SAML flows to include ``desktop_login`` where applicable.
 - Added new API endpoint ``GET`` ``/api/v4/channels/<channel-id>/common_teams`` to fetch list of teams common between members of a group message.
 - Added new API endpoint ``POST`` ``/api/v4/channels/<channel-id>/convert_to_channel`` to convert a group message to a private channel.
 - Added a new ``MessageHasBeenDeleted`` hook to the plugin API.
 - Moved the ``request`` package into the public shared folder.

### Go Version
 - v9.1 is built with Go ``v1.20.7``.

### Known Issues

>  - GM to channel conversion should show an error "A channel with that name already exists on the same team" [MM-54713](https://mattermost.atlassian.net/browse/MM-54713).

Reviewer 2: ```suggestion
 - Converting a group message to a channel should show an error "A channel with that name already exists on the same team" for duplicate channel names [MM-54713](https://mattermost.atlassian.net/browse/MM-54713).
```

>  - Marking a GM as unread does not resurface the numbered notification badge [MM-54778](https://mattermost.atlassian.net/browse/MM-54778).

Reviewer 2: ```suggestion
 - Marking a group message as unread doesn't resurface the numbered notification badge [MM-54778](https://mattermost.atlassian.net/browse/MM-54778).
```
